numba/cuda/cudamath.py

```diff
@@ -43,6 +43,7 @@ class Math_unary(ConcreteTemplate):
 @infer_global(math.log2)
 @infer_global(math.log10)
 @infer_global(math.exp)
+@infer_global(math.exp2)
```

Comment:
I think this will need a guard to ensure it doesn't break Python < 3.11.

(However, the CUDA changes should be taken out of this PR, and added to a PR in https://github.com/NVIDIA/numba-cuda) instead please.